### PR TITLE
Add Catx

### PR DESCRIPTION
**Pull request by cyprienc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by WissBe** [comment]
I am happy to be listed as a maintainer

**Comment by cyprienc** [comment]
Hello @conda-forge/help-python, this is ready for review!
Thanks.

**Comment by cyprienc** [comment]
Hello https://github.com/orgs/conda-forge/teams/help-python, this is ready for review!
Thanks.

File: recipes/catx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/catx-{{ version }}.tar.gz
+  sha256: 31c8b46ae6c8ffe39fe95ef40106e594780a95190ad83c13d28f22b295a4ae44
+
+build:
+  skip: True  # [win or py<37]
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:13]:
```suggestion
  entry_points:
    - project_name = project_name.__main__:main
  noarch: python
```
This allows the package to be noarch. The missing tensorflow dependency can be ignored as the solver will inform the user that it is not available during installation.

File: recipes/catx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:20]:
```suggestion
    - python >=3.7
```

File: recipes/catx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  run:
+    - chex
+    - dm-haiku
+    - jax
+    - matplotlib-base
+    - numpy
+    - optax
+    - python
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:28]:
```suggestion
    - python  >=3.7
```

File: recipes/catx/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - chex
+    - dm-haiku
+    - jax
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:24]: This is not listed upstream, are you sure that you need it?

File: recipes/catx/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - matplotlib-base
+    - numpy
+    - optax
+    - python
+    - tensorflow =2
+    - typing_extensions  # [py<310]
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:30]: This is also not listed upstream?

File: recipes/catx/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+  run:
+    - chex
+    - dm-haiku
+    - jax
+    - matplotlib-base
+    - numpy
+    - optax
+    - python
+    - tensorflow =2
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:29]: The version is not listed upstream. Would be good to have it there as well, if it is needed.

File: recipes/catx/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    - typing_extensions  # [py<310]
+
+test:
+  imports:
+    - catx
+  commands:
+    - pytest tests
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:36]:
```suggestion
    - pytest tests
    - pip check
    - project_name --help
```
This tests the dependency and the CLI.

File: recipes/catx/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  source_files:
+    - tests
+  requires:
+    - pytest
+    - pytest-mock
+    - scikit-learn
```
Comment (BastianZim) [inline recipes/catx/meta.yaml:42]:
```suggestion
    - scikit-learn
    - pip
```

**Review by BastianZim — COMMENTED** [review]
Just a couple of small things, then this is ready to go 🚀

**Pull request closed without merging**